bindings/python/nrm/client.py

```diff
@@ -0,0 +1,248 @@
+    "nrm_sensor_uuid", [nrm_sensor], nrm_str, Error.checkptr
+)
+
+nrm_slice_uuid = _nrm_get_function(
+    "nrm_slice_uuid", [nrm_slice], nrm_str, Error.checkptr
+)
+
+nrm_actuator_uuid = _nrm_get_function(
+    "nrm_actuator_uuid", [nrm_actuator], nrm_str, Error.checkptr
+)
+
+nrm_actuator_clientid = _nrm_get_function(
+    "nrm_actuator_clientid", [nrm_actuator], nrm_uuid, Error.checkptr
+)
+
+nrm_actuator_value = _nrm_get_function(
+    "nrm_actuator_value", [nrm_actuator], c_double, None
+)
+
+nrm_actuator_choices = _nrm_get_function(
+    "nrm_actuator_choices", [nrm_actuator], nrm_vector, Error.checkptr
+)
+
+
+class Client:
+    """Client class for interacting with NRM C interface.
+    Tentative usage:
+    ```
+    from nrm import Client, Actuator
+
+    my_client = Client("tcp://127.0.0.1", 2345, 3456)
+    ...
+    my_client.append_actuator(my_actuator)
+
+    """
+
+    def __init__(
+        self,
+        uri: str = None,
+        pub_port: int = None,
+        rpc_port: int = None,
+    ):
+        self.uri = uri if uri else upstream_uri
+        self.pub_port = pub_port if pub_port else upstream_pub_port
+        self.rpc_port = rpc_port if rpc_port else upstream_rpc_port
+        self.client = nrm_client(0)
+
+        if isinstance(self.uri, str):
+            self.uri = bytes(self.uri, "utf-8")
+
+        nrm_client_create(
+            byref(self.client), self.uri, self.pub_port, self.rpc_port
+        )
+
+    def list_sensors(self) -> list:
+        vector = nrm_vector(0)
+        nrm_client_list_sensors(self.client, byref(vector))
+        pylist = _nrm_vector_to_list_by_type(vector, nrm_sensor)
+        return [Sensor(i) for i in pylist]
+
+    def list_actuators(self) -> list:
+        vector = nrm_vector(0)
+        nrm_client_list_actuators(self.client, byref(vector))
+        pylist = _nrm_vector_to_list_by_type(vector, nrm_actuator)
+        return [Actuator(i) for i in pylist]
+
+    def list_scopes(self) -> list:
+        vector = nrm_vector(0)
+        nrm_client_list_scopes(self.client, byref(vector))
+        pylist = _nrm_vector_to_list_by_type(vector, nrm_sensor)
+        return [Scope(i) for i in pylist]
+
+    def list_slices(self) -> list:
+        vector = nrm_vector(0)
+        nrm_client_list_slices(self.client, byref(vector))
+        pylist = _nrm_vector_to_list_by_type(vector, nrm_slice)
+        return [Slice(i) for i in pylist]
+
+    def add_sensor(self, name: str):
+        sensor = nrm_sensor_create(bytes(name, "utf-8"))
+        nrm_client_add_sensor(self.client, sensor)
+
+    def add_actuator(self, name: str):
+        actuator = nrm_actuator_create(bytes(name, "utf-8"))
+        nrm_client_add_actuator(self.client, actuator)
+
+    def add_scope(self, name: str):
+        scope = nrm_scope_create(bytes(name, "utf-8"))
+        nrm_client_add_scope(self.client, scope)
+
+    def add_slice(self, name: str):
+        nslice = nrm_slice_create(bytes(name, "utf-8"))  # "slice" is builtin
```

Comment:
```suggestion
        slice = nrm_slice_create(bytes(name, "utf-8"))
```

Reply:
"slice" is a python builtin class. We shouldn't overwrite it